Interaction - Missions: Chains of Promathia Chapter 1

claywar opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

* Adds additional default dialogue changes and switches to Mission variables for Rites of Life (Reasoning behind using vars instead of missionStatus is to reserve space for once we decipher its impact on the mission log
* Audits, updates, and converts Below the Arks and The Mothercrystals to Interaction Framework.
* Implements the sealing of memories for Incomplete areas
* Implement memory sealing from left apparatus.
* Adds CoP helpers lua for reusing above implementation with both Below the Arks and Mothercrystals 
* Adds distance check to Right-side large apparatus (<= 2.71 yalms)

claywar committed:
Interaction - Mission: Rites of Life additional dialogue


Add Harith default action, and TODO for verification


Remove TODO after verification

zach2good commented on scripts/globals/bcnm.lua:
```diff
@@ -649,11 +649,11 @@ local function checkReqs(player, npc, bfid, registrant)
-        [ 768] = function() return ( (cop == mi.cop.BELOW_THE_ARKS and copStat==1) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_HOLLA))   ) end, -- PM1-3: The Mothercrystals
+        [ 768] = function() return ( (cop == mi.cop.BELOW_THE_ARKS) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_HOLLA))   ) end, -- PM1-3: The Mothercrystals
         [ 769] = function() return ( player:hasKeyItem(xi.ki.CENSER_OF_ABANDONMENT)                                                                                        ) end, -- ENM: Simulant
-        [ 800] = function() return ( (cop == mi.cop.BELOW_THE_ARKS and copStat==1) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_DEM))     ) end, -- PM1-3: The Mothercrystals
+        [ 800] = function() return ( (cop == mi.cop.BELOW_THE_ARKS) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_DEM))     ) end, -- PM1-3: The Mothercrystals
         [ 801] = function() return ( player:hasKeyItem(xi.ki.CENSER_OF_ANTIPATHY)                                                                                          ) end, -- ENM: You Are What You Eat
-        [ 832] = function() return ( (cop == mi.cop.BELOW_THE_ARKS and copStat==1) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_MEA))     ) end, -- PM1-3: The Mothercrystals
+        [ 832] = function() return ( (cop == mi.cop.BELOW_THE_ARKS) or (cop == mi.cop.THE_MOTHERCRYSTALS and not player:hasKeyItem(xi.ki.LIGHT_OF_MEA))     ) end, -- PM1-3: The Mothercrystals
```
Can we keep the formatting here?

zach2good commented on scripts/zones/Hall_of_Transference/npcs/_0e6.lua:
```diff
@@ -22,7 +22,7 @@ entity.onTrigger = function(player, npc)
         player:messageSpecial(ID.text.NO_RESPONSE_OFFSET + 6) -- Device seems to be functioning correctly.
     else
-        player:startEvent(167) -- Hexagonal Cones
+        player:startEvent(167) -- Hexagonal Cones -- FREAKING EVENT 165
```
?

zach2good commented on scripts/zones/Hall_of_Transference/npcs/_0e8.lua:
```diff
@@ -22,7 +22,7 @@ entity.onTrigger = function(player, npc)
         player:messageSpecial(ID.text.NO_RESPONSE_OFFSET + 6) -- Device seems to be functioning correctly.
     else
-        player:startEvent(163) -- Hexagonal Cones
+        player:startEvent(163) -- Hexagonal Cones -- 167 here
```
?

claywar committed:
Interaction - Mission: Below the Arks


Safety checkin for known issue


Reuse 32001 events, and break out shattered telepoints

Wishful thinking that they were all the same event, breaking out due to La Theine

Remove dead code


Fix bad copy-pasta


Start on CoP Mission Helper, restrict Monbereaux Dialogue


Fix Harith dialogue for Empty Memories quest


Correct Right-side apparatus events, style for bcnm lua

zach2good commented on scripts/missions/cop/1_3_The_Mothercrystals.lua:
```diff
@@ -0,0 +1,166 @@
+        {
+            onEventFinish =
+            {
+                [32001] = xi.cop.helpers.spireEventFinish,
```
All very neat and tidy 👍

claywar committed:
Interaction - Mission: The Mothercrystals


Implement additional sealing logic and cutscene adjustments

claywar commented:
I'm about as happy as I can be with this at the moment, appreciate feedback.

zach2good commented on scripts/missions/cop/1_3_The_Mothercrystals.lua:
```diff
@@ -0,0 +1,230 @@
+    nextMission = { xi.mission.log_id.COP, xi.mission.id.cop.AN_INVITATION_WEST },
+}
+
+-- Some helper functions require access to this mission class in order to operate.  The below
+-- functions wrap the helper to ensure that value gets to them.
+local shatteredTelepointSealMemory = function(player, csid, option, npc)
+    xi.cop.helpers.shatteredTelepointSealMemory(mission, player, csid, option, npc)
+end
+
+local largeApparatusOnTrigger = function(player, npc)
+    return xi.cop.helpers.largeApparatusOnTrigger(mission, player, npc)
+end
+
+local largeApparatusOnEventFinish = function(player, csid, option, npc)
+    xi.cop.helpers.largeApparatusOnEventFinish(mission, player, csid, option, npc)
+end
+
+local spireEventFinish = function(player, csid, option, npc)
+    xi.cop.helpers.spireEventFinish(mission, player, csid, option, npc)
+end
```
Good compromise :+1:

claywar committed:
QC: Chains of Promathia Chapter 1


Remove players from zone after BCNM complete if they have already completed mission


Pherimocial default event to DefaultActions (needed for replaceDefault to work)


QC: Wrap CoP mission helpers that require mission class


Wrap Mothercrystals functions, fix error


Style check fixes


Add additional check to Large Apparatus


Additional fixes for event IDs and offsets


Final QC round, ensure teleports, fix missing events


Add safety to ensure that helper is only called for relevant missions


Remove debug print


Add stopgap for mission variable
